This error is fixed in the kernel version 2.6.39, so this patch is for older kernels. "gcc" version<4.6 does not see this error. See the discussion http://www.kerneltrap.com/mailarchive/linux-netdev/2010/10/26/6288351 .
diff -u -r a/drivers/net/igbvf/igbvf.h b/drivers/net/igbvf/igbvf.h
--- a/drivers/net/igbvf/igbvf.h	2010-02-24 20:52:17.000000000 +0200
+++ b/drivers/net/igbvf/igbvf.h	2011-04-29 16:36:50.227295566 +0300
@@ -126,7 +126,6 @@
 			unsigned int page_offset;
 		};
 	};
-	struct page *page;
 };
 
 union igbvf_desc {
